### PR TITLE
Add spark job

### LINKED ISSUE #44: Install Spark Job
Install the Spark Job for dependency processing when Elasticsearch/cassandra is used.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #44 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/140)

**Comment by pavolloffay** [comment]
At the moment this does not change any UI configuration. Baed on the job configuration (enabled) and storage operator could enable/disable dependencies tab in UI. I will add it after this PR is merged.

**Comment by pavolloffay** [comment]
Also the job is by default disabled - we can enable it by default for supported storages.

File: deploy/examples/simplest.yaml [hunk 1]
```diff
@@ -2,3 +2,13 @@ apiVersion: io.jaegertracing/v1alpha1
 kind: Jaeger
 metadata:
   name: simplest
+  strategy: allInOne
```
Comment (pavolloffay) [inline deploy/examples/simplest.yaml]: I will remove this

**Comment by pavolloffay** [comment]
> Just need to adjust a few imports

I prefer to get merged #151 first.

**Comment by pavolloffay** [comment]
PR updated

**Comment by pavolloffay** [comment]
@jpkrohling could you please review this PR?

**Comment by pavolloffay** [comment]
> As mentioned above - I think it would be better to keep Spark out of the type and field names - its just an 

I am not sure if we could use that unknown future project with this interface - as the interface is highly dependent on the properties exposed by `spark-dependencies` project. For this reason I would keep it as with spark prefix.

**Comment by pavolloffay** [comment]
@jpkrohling could you please create a separate issue for any missing comments?

**Pull request merged**